carma_cooperative_perception/test/test_msg_conversion.cpp

```diff
@@ -391,3 +394,120 @@ TEST(ToExternalObjectList, FromTrackList)
+  
+  std::string proj_string{"+proj=tmerc +lat_0=42.24375605014171 +lon_0=-83.55739733422793 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=m +vunits=m +no_defs"};
+  auto shared_transform = std::make_shared<lanelet::projection::LocalFrameProjector>(proj_string.c_str());
+  const auto detected_object{
+    carma_cooperative_perception::to_detected_object_data_msg(object, shared_transform)};
+
+  EXPECT_EQ(detected_object.detected_object_common_data.obj_type.object_type, 1);
+  EXPECT_EQ(detected_object.detected_object_common_data.obj_type_cfd.classification_confidence, 90);
+  EXPECT_EQ(detected_object.detected_object_common_data.detected_id.object_id, 7);
+  EXPECT_NEAR(detected_object.detected_object_common_data.speed.speed, std::sqrt(481), 1e-2);
+  EXPECT_EQ(detected_object.detected_object_common_data.speed_z.speed, object.velocity_inst.twist.linear.z);
+
+  EXPECT_EQ(detected_object.detected_object_optional_data.det_veh.size.vehicle_width, 22);
+  EXPECT_EQ(detected_object.detected_object_optional_data.det_veh.size.vehicle_length, 21);
+  EXPECT_EQ(detected_object.detected_object_optional_data.det_veh.height.vehicle_height, 23);
+
+  // heading - not done
+}
+
+TEST(ToSDSMMsg, FromExternalObjectList)
+{
+  carma_perception_msgs::msg::ExternalObjectList object_list;
+  object_list.objects.emplace_back();
+  object_list.objects.emplace_back();
+
+  geometry_msgs::msg::PoseStamped current_pose;
+  current_pose.pose.position.x = 1;
+  current_pose.pose.position.y = 2;
+  current_pose.pose.position.z = 3;
+
+  lanelet::projection::LocalFrameProjector local_projector("+proj=tmerc +lat_0=39.46636844371259 +lon_0=-76.16919523566943 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=m +vunits=m +no_defs");
+  std::shared_ptr<lanelet::projection::LocalFrameProjector> map_projection;
+  map_projection = std::make_shared<lanelet::projection::LocalFrameProjector>(local_projector);
+
+  const auto sdsm{carma_cooperative_perception::to_sdsm_msg(object_list, current_pose, map_projection)};
+
+  std::cout << "SDSM year: " << sdsm.sdsm_time_stamp.year.year << std::endl;
+
+  EXPECT_EQ(std::size(sdsm.objects.detected_object_data), 2U);
+}
+
+TEST(ToWGSHeading, FromMapYaw)
+{
+
+  std::string proj_string{"+proj=tmerc +lat_0=42.24375605014171 +lon_0=-83.55739733422793 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=m +vunits=m +no_defs"};
+
+  // Test conversion from map to lat/lon
+  // Define map coordinates
+  lanelet::BasicPoint3d obj_map_coordinates;
+  obj_map_coordinates.x() = -69.7311856222;
+  obj_map_coordinates.y() = 331.419278969;
+  obj_map_coordinates.z() = 37.8485517148;
+
+  double yaw = 10.0;
+  auto shared_transform = std::make_shared<lanelet::projection::LocalFrameProjector>(proj_string.c_str());
+  double heading = carma_cooperative_perception::remove_units(carma_cooperative_perception::enu_orientation_to_true_heading(yaw, obj_map_coordinates, shared_transform));
+
+  EXPECT_EQ(heading, 80);
+
+}
+
+TEST(toSDSMMsg, getSDSMOffset)
```

Comment:
Same convention comment as above. Additionally, we should be using `PascalCase`.